Unexpected flush causes NPE in connection pool

Unpexxted flush causes NPE while doing a heartbeat.

```
2023-10-27 10:44:42 Caused by: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientStream.doStreamSignalNow(long, int)" because "stream" is null
2023-10-27 10:44:42     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientSignaler.signalNow(KafkaClientConnectionPool.java:489)
2023-10-27 10:44:42     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory$CoordinatorClient.doHeartbeatRequest(KafkaClientGroupFactory.java:3949)
2023-10-27 10:44:42     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory$KafkaGroupStream.onStreamFlush(KafkaClientGroupFactory.java:1413)
2023-10-27 10:44:42     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory$KafkaGroupStream.onStream(KafkaClientGroupFactory.java:1297)
2023-10-27 10:44:42     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientGroupFactory.lambda$newStream$1(KafkaClientGroupFactory.java:366)
2023-10-27 10:44:42     at 
```

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
@@ -3946,7 +3946,10 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
         private void doHeartbeatRequest(
             long traceId)
         {
-            if (!members.isEmpty())
+            final String memberId = delegate.groupMembership.memberIds.getOrDefault(delegate.groupId, UNKNOWN_MEMBER_ID);
+
+            if (KafkaState.initialOpened(state) &&
+                !memberId.equals(UNKNOWN_MEMBER_ID))
             {
                 if (heartbeatRequestId != NO_CANCEL_ID)
                 {

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
@@ -3946,9 +3946,7 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
         private void doHeartbeatRequest(
             long traceId)
         {
-            final String memberId = delegate.groupMembership.memberIds.getOrDefault(delegate.groupId, UNKNOWN_MEMBER_ID);
-
-            if (!memberId.equals(UNKNOWN_MEMBER_ID))
+            if (!members.isEmpty())
             {
                 if (heartbeatRequestId != NO_CANCEL_ID)
                 {

